### PR TITLE
[FLINK-25545][flink-clients][JUnit5 Migration] Module: flink-clients.

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

[JUnit5 Migration] Module: flink-clients. 

## Brief change log

[JUnit5 Migration] Module: flink-clients. 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
Hi, @fapaul Would  you mind helping to check this flink-clients JUnit5 Migration ? Thank you very much.

**Comment by fapaul** [comment]
@zentol do you have time to look at this?

**Comment by RocMarshal** [comment]
@RyanSkraba Thank you very much for the wonderful review. I'll advance it based on your comments ASAP.

File: flink-clients/src/test/java/org/apache/flink/client/ExecutionEnvironmentTest.java [hunk 1]
```diff
@@ -26,14 +26,15 @@
...
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
 import java.io.Serializable;
 
-import static junit.framework.TestCase.fail;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
```
[the remark this answers is not in the document]
Comment (RocMarshal) [reply flink-clients/src/test/java/org/apache/flink/client/ExecutionEnvironmentTest.java:37]: Yes. I'll update it.

**Comment by zentol** [comment]
@RocMarshal Please be aware of this nifty migration tool that francesco has written: https://github.com/slinkydeveloper/assertj-migrator
It can take care of a large number of asserion migrations.

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
Thanks for the @RyanSkraba  review and @zentol  research.
I updated it based on your suggestions. PTAL.

**Review by RyanSkraba — APPROVED** [review]
LGTM!

I left a couple of comments where there _could_ be some additional clean-up, but there is nothing broken and this could/should be merged as is!  I don't want to be that guy with "just one more thing..." :D 

There were a lot of lines of code changed here and some of it was pretty finicky, thanks so much for doing such a great job!

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Review by gaoyunhaii — APPROVED** [review]
Very thanks @RocMarshal for the PR and everyone for the review! Also LGTM

**Pull request closed without merging**